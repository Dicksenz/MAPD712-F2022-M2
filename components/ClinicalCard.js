import React from "react";
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";

import Icon from "@expo/vector-icons/MaterialCommunityIcons";
import { Colors } from "react-native/Libraries/NewAppScreen";

const ClinicalCard = ({ category, date, readings, nurse }) => {
  return (
    <View style={styles.container}>
      <View style={{ display: "flex", flexDirection: "row" }}>
        <View style={styles.nameContainer}>
          <Text style={{ ...styles.nameTextStyle, marginBottom: 10 }}>
            {category}
          </Text>

          {category === "Blood Pressure" && (
            <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
              Blood pressure value: {readings.systolic} / {readings.diastolic}{" "}
              mmHg
            </Text>
          )}
          {category === "Respiratory rate" && (
            <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
              Respiratory Rate value: {readings.bpm} BPM
            </Text>
          )}
          {category === "Blood Oxygen Level" && (
            <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
              Blood Oxygen level: {readings.percentage} %
            </Text>
          )}
          {category === "Heart Beat Rate" && (
            <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
              Heart Beat Rate value: {readings.bpm} BPM
            </Text>
          )}

          <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
            Taken by nurse: {nurse}
          </Text>
          <Text style={{ ...styles.subtitleStyle, marginBottom: 10 }}>
            Date tested: {date}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "white",
    paddingTop: 20,
    paddingBottom: 20,
    paddingLeft: 10,
    paddingRight: 10,
    marginBottom: 10,
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 2,
    elevation: 2,
    borderRadius: 15,
  },
  nameContainer: {
    display: "flex",
    flexDirection: "column",
  },
  nameTextStyle: {
    color: "#C4CD4A",
    fontSize: 20,
    fontWeight: "bold",
  },
  subtitleStyle: {
    color: "#616161",
    fontSize: 18,
  },

  avatar: {
    backgroundColor: "#F9F9F9",

    borderColor: "",
    padding: 10,
    borderRadius: 50,
    justifyContent: "center",
    alignItems: "center",
    marginRight: 10,
  },

  arrowCircle: {
    backgroundColor: "#347174",
    height: 30,
    width: 30,
    borderRadius: 50,
  },
});

export default ClinicalCard;
