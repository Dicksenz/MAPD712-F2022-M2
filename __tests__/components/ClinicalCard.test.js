import React from "react";
import renderer from "react-test-renderer";
import ClinicalCard from "../../components/ClinicalCard";

test("Check if ClinicalCard component renders correctly", () => {
  const tree = renderer.create(<ClinicalCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
