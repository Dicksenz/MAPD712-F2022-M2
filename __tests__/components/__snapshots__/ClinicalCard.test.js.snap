// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if ClinicalCard component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 15,
      "display": "flex",
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "paddingBottom": 20,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 20,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#C4CD4A",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 10,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#616161",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Taken by nurse: 
      </Text>
      <Text
        style={
          Object {
            "color": "#616161",
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        Date tested: 
      </Text>
    </View>
  </View>
</View>
`;
